extensions/pyRevitTools.extension/pyRevit.tab/Analysis.panel/Tools.stack/Analyse.pulldown/Get Centroid.pushbutton/script.py

```diff
@@ -0,0 +1,94 @@
+from __future__ import print_function
+from pyrevit import script, revit, DB
+
+doc = revit.doc
+uidoc = revit.uidoc
+
+logger = script.get_logger()
+output = script.get_output()
+
+
+def merge_solids(solids):
+    """Merges the provided list of solids into a single unified solid
+
+    Args:
+        solids (list(DB.Solid)): The solids to merge
+
+    Raises:
+        ValueError: If the input is an empty list
+
+    Returns:
+        DB.Solid: The union of the solids
+    """
+    union = None
+    if not solids:
+        raise ValueError('List of provided solids was empty.')
+    for solid in solids:
```

Comment:
You may need a try except somewhere here as it does not process to merging all the solids and when it cannot it throws:
```
File "C:\pyRevit\B1_ToolBar.extension\Outils BIM One.tab\test.panel\test.pulldown\z.pushbutton\script.py", line 49, in <module>
 File "C:\pyRevit\B1_ToolBar.extension\Outils BIM One.tab\test.panel\test.pulldown\z.pushbutton\script.py", line 27, in merge_solids
Exception: Failed to perform the Boolean operation for the two solids. This may be due to geometric inaccuracies in the solids, such as slightly misaligned faces or edges. If so, eliminating the inaccuracies by making sure the solids are accurately aligned may solve the problem. This also may be due to one or both solids having complexities such as more than two faces geometrically meeting along a single edge, or two coincident edges, etc. Eliminating such conditions, or performing a sequence of Boolean operations in an order that avoids such conditions, may solve the problem.
```

Reply:
other than that, I love it!

Reply:
You're right, I've already encountered that issue before in other tools. I think this might be related to the limitation that you can't have edges shorter than ~0.7mm, and the inaccuracies can produce such edges on the joined geometries, so it throws this exception.
I will undo the whole process, and try to provide some details on the problematic elements.

Reply:
I've changed the code, please have a look. Unfortunately the reference extracted from the solid points to the element's type, so reporting the id is not useful...